src/main/java/de/tum/in/www1/artemis/domain/plagiarism/PlagiarismResult.java

```diff
@@ -18,19 +50,25 @@
     protected long duration;
 
     /**
-     * ID of the exercise for which plagiarism was detected.
+     * Exercise for which plagiarism was detected.
      */
-    protected long exerciseId;
+    @JsonIdentityReference(alwaysAsId = true)
+    @JsonProperty("exerciseId")
+    @ManyToOne
+    protected Exercise exercise;
```

Comment:
It would be perfectly find to load the exercise here and to use a simpler mechanism